### PR TITLE
Initial converge tests for AWS

### PR DESCRIPTION
**Pull request by 14rcole**
`molecule converge` tests the ansible roles to make sure they function correctly.  This PR can serve as the basis for convergence tests for all of the roles.

Given that we mock the test data, I feel like we can just have a a resource definition for each role within the provider (e.g. ec2, ec2_vpc_net, etc).  Any reason why this can't work?  If not I'll stack another commit on top of this that contains all of the AWS tests

### CODE REVIEW

File: linchpin/provision/roles/aws/tasks/teardown_resource_group.yml [hunk 1]
```diff
@@ -23,6 +23,13 @@
   when: auth_var is defined
   no_log: "{{ not auth_debug }}"
 
+- shell:
```
Comment (samvarankashyap) [inline linchpin/provision/roles/aws/tasks/teardown_resource_group.yml]: @14rcole Do we need these debug statements?

Comment (14rcole) [reply linchpin/provision/roles/aws/tasks/teardown_resource_group.yml]: Nope, thank you

**Comment by 14rcole** [comment]
@samvarankashyap I'm still getting failures from two of the tests because of missing mock data, but can we merge this and uncomment those after we get aws access back?

**Comment by samvarankashyap** [comment]
@14rcole As long as we make sure we have an issue to track it. We can do that.

**Comment by samvarankashyap** [comment]
@14rcole Before you merge it can we have an issue that points out the problem this PR solves. 
Thank you

**Comment by 14rcole** [comment]
Fixes #1619

**Comment by 14rcole** [comment]
[merge]

**Pull request merged**